src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -328,6 +333,9 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
 
             // Check if package name matches regex
             Matcher packageNameMatcher = packageNamePattern.matcher(programmingExercise.getPackageName());
+            if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.SWIFT) {
+                packageNameMatcher = packageNamePatternForSwift.matcher(programmingExercise.getPackageName());
+            }
```

Comment:
It probably isn't useful to match on the regular package name Regex first if we're dealing with a Swift exercise. What do you think about changing this to something like
```
Matcher packageNameMatcher = null;
if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.SWIFT) {
    packageNameMatcher = packageNamePatternForSwift.matcher(programmingExercise.getPackageName());
} else {
    packageNamePattern.matcher(programmingExercise.getPackageName());
}
```
?

Reply:
Good suggestion!

Reply:
@FrankeLukas your suggestion is basically the same but more lines of code, right? In case we don't have a Swift Exercise, the packageNameMatcher gets assigned the packageNamePattern in the else part. To save this else part I am assigning it always instead of first setting it to null.

Reply:
but I guess you're right. I've fixed it :)